src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -60,35 +69,88 @@ public static String generateClassBeanName(Class<?> beanClass) {
-        .map(Field::getType);
+  /**
+   * Finds and returns the initialization constructor for a given bean class. The initialization
+   * constructor is determined based on the presence of the {@link Autowired} annotation and the
+   * number of available constructors.
+   *
+   * @param beanClass the class of the bean to find the initialization constructor for.
+   * @param beanName  the name of the bean, used for logging and error handling.
+   * @return the initialization constructor for the specified bean class.
+   * @throws BeanDefinitionCreationException If there are issues with the bean's constructors, such
+   *                                         as having more than one constructor without the
+   *                                         {@link Autowired} annotation or multiple constructors
+   *                                         marked with {@link Autowired}.
+   */
+  public static Constructor<?> findBeanInitConstructor(Class<?> beanClass, String beanName) {
+    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
 
-    Stream<Class<?>> methodDependenciesStream = Arrays.stream(beanClass.getDeclaredMethods())
-        .filter(method -> method.isAnnotationPresent(Autowired.class))
-        .flatMap(method -> Arrays.stream(method.getParameterTypes()));
+    // One constructor with or without @Autowired.
+    if (declaredConstructors.length == 1) {
+      log.trace("One init constructor found and registered [{}] for bean candidate [{}]",
+          declaredConstructors[0], beanClass.getName());
+      return declaredConstructors[0];
+    } else if (declaredConstructors.length > 1) {
+      log.trace("Multiple constructors found for bean candidate [{}]", beanClass.getName());
+      if (isAnnotationPresentForAnyConstructor(Autowired.class, declaredConstructors)) {
+        // Multiple constructors with only one @Autowired
+        // TODO: checked
+        if (isAnnotationPresentForSingleConstructorOnly(Autowired.class, declaredConstructors)) {
+          Constructor<?> initConstructor =
+              getConstructorsAnnotatedWith(Autowired.class, declaredConstructors).get(0);
+          log.trace("@Autowired constructor found for bean candidate [{}]: [{}]",
+              beanClass.getName(),
+              initConstructor);
+          return initConstructor;
+        }
+        // TODO: checked
+        // Multiple constructors with multiple @Autowired annotations
+        List<Constructor<?>> autowiredConstructors =
+            getConstructorsAnnotatedWith(Autowired.class, declaredConstructors);
+        log.error(
+            "Bean candidate [{}] of type [{}] has more then 1 constructor marked with @Autowired.",
+            beanName, beanClass);
+        throw new BeanDefinitionCreationException(MULTIPLE_AUTOWIRED_CONSTRUCTORS_MESSAGE
+            .formatted(beanName, autowiredConstructors.get(1), autowiredConstructors.get(0)));
```

Comment:
looks like there possibly more than 2 constructors with this annotation maybe we will just return the count of contractors

Reply:
I believe it's a matter of taste in how we inform users about multiple autowired constructors. In this case, it says that the second is not valid because we already found one that is the first one.
Personally, I would say that both the current option and the option you propose are equally informative, so I don't see a big reason to change the error message.